### PR TITLE
soft release introduce new data structure for front end

### PR DESCRIPTION
**Pull request by elisechant**

### CODE REVIEW

**Comment by elisechant** [comment]
requires PD-500 to be merged first.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+  end
+
   # Sets the body class on a per-page basis
   # Params:
   # - class_names: String of class names to append to class names string
   def body_class(class_names)
-    class_name_str = self.controller_name + '_' + self.action_name
+    class_name_str = body_route
     if !class_names.empty?
       class_name_str << ' ' + class_names
```
Comment (tobyhede) [inline app/helpers/application_helper.rb]: Could probably drop the assignement and just ` body_route <<`

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -24,7 +22,7 @@
   <%= render 'layouts/analytics' %>
 </head>
 
-<body class="<%= body_class(yield(:body_class_names)) %>">
+<body class="<%= body_class(yield(:body_class_names)) %>" data-route="<%= body_route %>">
```
Comment (tobyhede) [inline app/views/layouts/application.html.erb]: This smells funny. 
You should be able to reuse content_for in the helper

```
def body_class
  class_names = content_for(:body_class_names)
```

**Comment by tobyhede** [comment]
What does this actually do?

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (elisechant) [inline app/helpers/application_helper.rb]: yep, that's in the other ticket. Getting to it

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (elisechant) [inline app/views/layouts/application.html.erb]: i only want the route here, not all the classes like high_contrast_mode etc

**Comment by elisechant** [comment]
@tobyhede porting the gulp linear javascript in to webpack land so we can work with it (slightly better structured but no dramatic change to original codes)

**Pull request merged**